src/blocks/scratch3_motion.js

```diff
@@ -51,24 +52,31 @@ class Scratch3MotionBlocks {
         util.target.setXY(x, y);
     }
 
-    goTo (args, util) {
+    getTarget (TO, util) {
         let targetX = 0;
         let targetY = 0;
-        if (args.TO === '_mouse_') {
+        if (TO === '_mouse_') {
             targetX = util.ioQuery('mouse', 'getX');
             targetY = util.ioQuery('mouse', 'getY');
-        } else if (args.TO === '_random_') {
+        } else if (TO === '_random_') {
             const stageWidth = this.runtime.constructor.STAGE_WIDTH;
             const stageHeight = this.runtime.constructor.STAGE_HEIGHT;
             targetX = Math.round(stageWidth * (Math.random() - 0.5));
             targetY = Math.round(stageHeight * (Math.random() - 0.5));
         } else {
-            const goToTarget = this.runtime.getSpriteTargetByName(args.TO);
+            const goToTarget = this.runtime.getSpriteTargetByName(TO);
             if (!goToTarget) return;
             targetX = goToTarget.x;
             targetY = goToTarget.y;
         }
-        util.target.setXY(targetX, targetY);
+        return [targetX, targetY];
+    }
+
+    goTo (args, util) {
+        const targetXY = getTarget(args.TO, util);
```

Comment:
`getTarget` isn't defined here (you noticed that!). See, the issue is that JavaScript expects there to be a variable named `getTarget`, and there isn't; it can't find any `let`, `var`, or `const getTarget`.

What *you* expected is that by declaring a *method* named `getTarget`, JavaScript would automatically know that "`getTarget`" means "the method named `getTarget`".

But JavaScript doesn't know that.

There's a specific syntax for getting a method, and it's actually the same as the syntax to get any other property: `this.methodName`. (You see, methods are (basically) just properties of the class instance, just like `this.runtime`.)

(In fact, you can already see `this.methodName` being used a bunch in this very file -- check the `getPrimitives` function, and you'll see that, for example, the line `motion_glideto: this.glideTo` is getting the method `glideTo` of `this` (so, the class instance) and storing it in the `motion_glideto` property of the return object!)

So, what you want here is `this.getTarget(args.TO, util)`, not `getTarget(args.TO, util)`.